src/devices/gridstream.c

```diff
@@ -0,0 +1,258 @@
+/** @file
+    Decoder for Gridstream RF devices produced by Landis & Gyr.
+
+    Copyright (C) 2016 Benjamin Larsson
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+ */
+
+/**
+Landis & Gyr Gridstream Power Meters.
+
+- Center Frequency: 915 Mhz
+- Modulation: FSK-PCM
+- Bitrates: 9600, 19200, 38400
+- Preamble: 0xAAAA
+- Syncword v4: 0b0000000001 0b0111111111
+- Syncword v5: 0b0000000001 0b11111111111
+
+Datastream is variable length and bitrate depending on type fields
+Preamble
+Bytes after preamble are encoded with 8N1
+Data layouts:
+    Subtype 55:
+        AAAAAA SSSS TT YY LLLL KK BBBBBBBBBB WWWWWWWWWW II MMMMMMMM KKKK EEEEEEEE KKKK KKKKKK CCCC KKKK XXXX KK
+    Subtype D2:
+        AAAAAA SSSS TT YY LL K----------K XXXX
+    Subtype D5:
+        AAAAAA SSSS TT YY LLLL KK DDDDDDDD EEEEEEEE II K----------K CCCC KKKK XXXX
+- A - Preamble
+- S - Syncword
+- T - Type
+- Y - Subtype
+- L - Length
+- B - Broadcast
+- D - Dest Address
+- E - Source Address
+- M - Uptime (time since last outage in seconds)
+- I - Counter
+- C - Clock
+- K - Unknown
+- X - CRC (poly 0x1021, init set by provider)
+
+*/
+
+#include "decoder.h"
+#include <stdbool.h>
+#include <stdlib.h>
+#include <time.h>
+
+static uint16_t known_crc_init[] = {
+        0xe623,
+        0x5fd6,
+        0xD553,
+        0x45F8,
+        0x62C1,
+        0x23D1,
+        0x2C22,
+        0x142A,
+};
+
+static int gridstream_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    data_t *data;
+    uint8_t const preamble[] = {
+            0xAA,
+            0xAA,
+            0x00,
+    };
+    uint8_t b[BITBUF_COLS];
+    uint16_t stream_len;
+    uint16_t crc;
+    uint16_t crc_count = 0;
+    bool crc_ok        = false;
+    char destwanaddress_str[13];
+    char srcwanaddress_str[13];
+    char srcaddress_str[9];
+    char destaddress_str[9];
+    uint32_t uptime;
+    time_t clock;
+    char clock_str[80];
+    int subtype;
+    unsigned offset = bitbuffer_search(bitbuffer, 0, 0, preamble, 24) + 16;
```

Comment:
Eventually we want to know if it is a V4 or V5 packet. You should use the sync words from the wiki and then align the bits by adjusting the offset manually depending on the version. But the comment below does not clearly indicate if this decoder also supports V5 packets.

Reply:
https://github.com/merbanan/rtl_433/pull/2616/commits/7edd93a5aead3c669f4bd8e5dbf2d6d32e9ede96 added handling for both V4 and V5 and verified with testing that both are being seen and decoded, in live and saved samples.